crates/noirc_frontend/src/hir/def_collector/dc_mod.rs

```diff
@@ -54,15 +57,149 @@ pub fn collect_defs(
 
     collector.collect_globals(context, ast.globals, errors);
 
+    collector.collect_traits(ast.traits, crate_id, errors);
+
     collector.collect_structs(ast.types, crate_id, errors);
 
     collector.collect_type_aliases(context, ast.type_aliases, errors);
 
     collector.collect_functions(context, ast.functions, errors);
 
+    collector.collect_trait_impls(context, ast.trait_impls, errors);
+
     collector.collect_impls(context, ast.impls);
 }
 
+fn check_trait_method_implementation_generics(
+    _generics: &[Ident],
+    _noir_function: &NoirFunction,
+    _trait_name: &str,
+) -> Result<(), DefCollectorErrorKind> {
+    // TODO
+    Ok(())
+}
+
+fn check_trait_method_implementation_parameters(
+    parameters: &Vec<(Ident, UnresolvedType)>,
```

Comment:
These are the parameters from the trait definition, right? If so, can we rename them to `expected_parameters` so it is more clear they are not the parameters of the impl function?

Reply:
done